plugins/meteor_engage_warn.py

```diff
@@ -75,9 +77,26 @@ def _display_meteor_pos_warning_msg(self, end_pos: Dict[str, float]) -> bool:
         pos_str = "\n". join(pos_str)
 
         # Guess (back) which position the users wants to go to
-        target_pos = self.main_app.main_frame.getCurrentPostureLabel(end_pos)
+        target_pos = self.main_app.main_data.posture_manager.getCurrentPostureLabel(end_pos)
+        stage = self.main_app.main_data.get_component("stage")
+        current_angle = 0
+        fav_angle = 0
+        angle = ''
         if target_pos == FM_IMAGING:
             warn_msg = FM_WARN_MSG
+            # Check the deviation in rotation angle when
+            # switching from SEM to FM,
+            # give a warning message if switching is done from a different rotation angle
+            stage_md = stage.getMetadata()
+            fav_angles = stage_md[model.MD_FAV_SEM_POS_ACTIVE]
+            for angle in ("rz", "rm"):
+                if angle in fav_angles:
+                    current_angle = stage.position.value[angle]
+                    fav_angle = fav_angles[angle]
+                    break
+            if current_angle != fav_angle:
```

Comment:
Wouldn't directly check floats like this, insteasd would use math.isclose() / np.isclose(). Otherwise it can fail for very small numerical differences